Python/optimizer_bytecodes.c

```diff
@@ -846,6 +846,10 @@ dummy_func(void) {
        next = sym_new_type(ctx, &PyLong_Type);
     }
 
+    op(_CALL_TYPE_1, (callable, null, arg -- res)) {
+        res = sym_new_type(ctx, &PyType_Type);
```

Comment:
Unfortunately, I don't think this is correct when metaclasses are involved:

```py
>>> class T(type):
...     pass
...     
>>> class C(metaclass=T):
...     pass
...     
>>> c = type(C())
>>> c.__class__ is type
False
```

In this case, the JIT would mistakenly think that `c.__class__` *is* `type`, which isn't true.

However, I think we can do better. If we know the type of `arg` (`sym_has_type`), then we can set the output to be a constant, that class (`sym_get_type`). Otherwise, we just need to create a new non-null symbol.

Reply:
Updated, I'll try to come up with more tests for it.